### ISSUE TITLE
EOS API: Panic error in parsing block

### ISSUE THREAD

**Issue by jackami** [eoscanada/eos-go#105]
When i parse the No.29750080 block, have a panic.

```
package eos_test

import (
	"fmt"
	"github.com/eoscanada/eos-go"
	_ "github.com/eoscanada/eos-go/system"
	_ "github.com/eoscanada/eos-go/token"
	"testing"
)

func TestGetBlockByNum(t *testing.T) {
	api := eos.New("https://nodes.get-scatter.com")
	blk, err := api.GetBlockByNum(29750080)
	if err != nil {
		t.Error(err.Error())
	} else {
		for _, action := range blk.Transactions {
			action.Transaction.Packed.Unpack()
		}
	}
}
```
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x12a3f1a]

goroutine 19 [running]:
testing.tRunner.func1(0xc000110200)
	/usr/local/go/src/testing/testing.go:792 +0x387
panic(0x13020e0, 0x16e7f70)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/eoscanada/eos-go.(*PackedTransaction).unpack(0x0, 0xc000096701, 0x8, 0x12e7ee0, 0xc000096730)
	/Users/eagle/Go/pkg/mod/github.com/eoscanada/eos-go@v0.8.0/transaction.go:238 +0x3a
github.com/eoscanada/eos-go.(*PackedTransaction).UnpackBare(0x0, 0x1c5f340, 0xc00028a000, 0x0)
```

**Comment by billettc** [comment]
Please see the following code.

You need to validate that the current transaction `trx.Transaction.Packed` is not nil. Not all transactions inside a given block has packed data.

PTW. By investigating your issue I did found  a log producing a `panic`. Please make sure to pull latest code 

```
package main

import (
	"fmt"
	"github.com/eoscanada/eos-go"
	_ "github.com/eoscanada/eos-go/system"
	_ "github.com/eoscanada/eos-go/token"
)

func main() {

	api := eos.New("https://nodes.get-scatter.com")
	api.Debug = true

	blk, err := api.GetBlockByNum(29750080)
	if err != nil {
		panic(err)
	} else {
		for _, trx := range blk.Transactions {
			packedTrx := trx.Transaction.Packed
			if packedTrx != nil {
				fmt.Println("Found a packed transaction:", trx.Transaction.ID)
				unpackTrx, err := packedTrx.Unpack()
				if err != nil {
					panic(err)
				}
				fmt.Println("unpacked trx with action count:", len(unpackTrx.Actions))
			} else {
				fmt.Println("No packed transaction found:", trx.Transaction.ID)
			}
		}
	}
}```

**Comment by billettc** [comment]
Fix: https://github.com/eoscanada/eos-go/commit/82abf1f7543efaf9e194fcc71c5f7a541e844afb

**Comment by jackami** [comment]
Thanks for your answer first。

My test code did not add the nil judgment，just a example。But the exception I said is indeed not caused by the nil problem. 

I debug this code and found that there is a problem with Decode.  I saw your problem fixed and i'll pull the latest code to resolve my problem.

[Fix: 82abf1f]
